### PR TITLE
Complete switch to dune

### PR DESCRIPTION
**Pull request by hra687261**
Removed Ocp-build's makefile and the standalone tests, reindented some files

### CODE REVIEW

**Comment by Stevendeo** [comment]
Can you postpone your indentation changes? Rebasing my PR is painful because of it, I will do a whole indentation pass in mine.

File: .gitignore [hunk 1]
```diff
@@ -16,5 +16,4 @@ config.status
 src/coreSig.ml
 src/extSigs.ml
```
Comment (Stevendeo) [inline .gitignore]: You can remove those two now they are not generated

File: tests/dune [hunk 2]
```diff
@@ -11,15 +11,3 @@
...
- (names solveEmpty standalone_minimal_maximization standalone_minimal)
- (modules
-  solveEmpty
-  standalone_minimal_maximization
-  standalone_minimal
-  rat
-  var
-  ex
-  instantiateBasicFunctor)
- (libraries ocplib-simplex))
```
Comment (Stevendeo) [inline tests/dune:25]: Why did you remove these files? They are useful for testing, they allow to quickly write simplex examples.

File: .gitignore (hunk 1 above)
Comment (hra687261) [reply .gitignore]: Ok

File: tests/dune (hunk 2 above)
Comment (hra687261) [reply tests/dune:25]: Their content was moved into [tests/mail.ml](https://github.com/hra687261/ocplib-simplex/blob/1ef855e705584d0ec8625c21e2ba2779067c33b2/tests/main.ml). That way dune's testing system can be used, and it's easier to see differences with the previous testing results (through [tests/main.expected](https://github.com/hra687261/ocplib-simplex/blob/1ef855e705584d0ec8625c21e2ba2779067c33b2/tests/main.expected)), and to update the expected result with dune promote.

**Comment by hra687261** [comment]
@Stevendeo Ok, I reverted the reindentation.

File: tests/dune (hunk 2 above)
Comment (Stevendeo) [reply tests/dune:25]: The dune testing system was already working before, wasn't it ? Actually, I think the files you removes should be part of a library that would be used by `main` and other future tests. In the current state of the project, there is only one test started by dune, which is not enough ; if we want to add more non-regression tests, it would be simpler to already have the functor instanciated.

Comment (hra687261) [reply tests/dune:25]: What was being tested before the last commit was the tests that were in main.ml, it actually contained the three tests that were in the three files that were deleted, but rewritten as functions. I wrote them that way to have one executable instead of three, but I guess it's more readable if I separate them back into three files.
Currently, the functor is instantiated inside the simplex.ml file. And the three test files use that same functor to run the tests.
To add more tests, you'd need to add an executable, add the test's name to the `tests` stanza inside the dune file, and add the corresponding rule.
